### PR TITLE
Change lxd to lxc for legacy juju

### PR DESCRIPTION
**Pull request by makyo**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-bundlelib/78/
Test PASSed.

File: jujubundlelib/changeset.py [hunk 1]
```diff
@@ -276,6 +279,10 @@ def _handle_container_placement(changeset, placement, machine_record_id):
     return container_record_id
 
 
+def _lxd_to_lxc(container_type):
+    return re.sub(r'^lxd$', 'lxc', container_type)
```
Comment (jrwren) [inline jujubundlelib/changeset.py]: I'd prefer to not even use re here.

    return 'lxc' if container_type=='lxd' else container_type

File: jujubundlelib/changeset.py [hunk 2]
```diff
@@ -218,7 +219,9 @@ def _handle_unit_placement(
             parent_record_id = 'addMachines-{}'.format(changeset.next_action())
             options = {}
             if placement.container_type:
-                options = {'containerType': placement.container_type}
+                options = {
+                    'containerType': _lxd_to_lxc(placement.container_type)
```
Comment (frankban) [inline jujubundlelib/changeset.py]: Why not just:
```python
# Define Linux container types.
LXC = 'lxc'
LXD = 'lxd'
```
and then, here 
```python
if placement.container_type:
    ctype = placement.container_type
    if ctype == LXD:
        ctype = LXC
    options = {'containerType': ctype}
```
I am not sure it's worth using regexp for such a simple substitution (probably `'lxd'.replace('lxd', 'lxc')` would be faster than regexp in this case, fwiw).

**Comment by mhilton** [comment]
:+1:

**Comment by jrwren** [comment]
👍 One readability preference. Take it or leave it.

**Comment by frankban** [comment]
Thank you Madison! 👍 with removing regexps.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-bundlelib/79/
Test PASSed.

**Comment by makyo** [comment]
Thanks all :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-bundlelib-merge

**Pull request merged**